runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/engine/internal/HttpClientMetrics.kt

```diff
@@ -25,8 +25,13 @@ public object HttpClientMetricAttributes {
     public val AcquiredConnection: Attributes = attributesOf { "state" to "acquired" }
     public val QueuedRequest: Attributes = attributesOf { "state" to "queued" }
     public val InFlightRequest: Attributes = attributesOf { "state" to "in-flight" }
+    public val TotalThread: Attributes = attributesOf { "state" to "total" }
```

Comment:
**correctness**:  The state attribute for this should probably follow the [usage](https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-naming) convention where the sub states total to some limit. 
e.g. thread state usage metric `smithy.client.http.threads.usage` with states of `idle | consumed` and a separate metric for the limit `smithy.client.http.threads.limit`.

Also I suppose we should probably get this defined in the SRA before codifying it in an implementation.

We may not know the total thread limit in every case or thread pools may be dynamic and spin down threads which would break the "should" part of the convention but I think that's fine. In the ideal case the convention applies and even when it doesn't it's still how you'd want it graphed.

Reply:
Sure, I'll change this to idle/active threads.